components/tools/OmeroWeb/omeroweb/webgateway/views.py

```diff
@@ -1247,6 +1250,76 @@ def render_col_plot(request, iid, z, t, x, w=1, conn=None, **kwargs):
 
 @login_required()
 @jsonp
+def api_container_list(request, conn=None, **kwargs):
+    # Get parameters
+    try:
+        page = getIntOrDefault(request, 'page', 1)
+        limit = getIntOrDefault(request, 'limit', settings.PAGE)
+        group_id = getIntOrDefault(request, 'group', -1)
+        experimenter_id = getIntOrDefault(request, 'id', -1)
+    except ValueError:
```

Comment:
Yes, I agree. This is just the current code from webclient ported over. I was wondering if there's a nice way to handle all these parameters without a ton of try / except blocks. But perhaps several try / except blocks is the clearest way of handling this.
In any case, I think it is preferable to raise an exception with invalid parameters rather than silently use the default - otherwise bug finding etc becomes much harder